### PR TITLE
✨ Configurable qos resources for containers managed by cluster-manager and klusterlet

### PR DESCRIPTION
**Pull request by promid**
## Summary
Configurable qos resources for containers managed by cluster-manager and klusterlet
## Related issue(s)

Fixes #

## Related PR
https://github.com/open-cluster-management-io/api/pull/316

### CODE REVIEW

**Comment by promid** [comment]
/hold
Pending https://github.com/open-cluster-management-io/api/pull/316 this PR

File: pkg/common/helpers/clusters_test.go [hunk 1]
```diff
@@ -1,114 +0,0 @@
-package helpers
```
Comment (haoqing0110) [inline pkg/common/helpers/clusters_test.go:1]: The `PlacementDecisionGetter` is being used by other controllers, `GetClusterChanges` and the clusters_test.go is used to test the `PlacementDecisionGetter` I think.

Comment (promid) [reply pkg/common/helpers/clusters_test.go:1]: I removed this because functions used in pkg/common/helpers/clusters.go are not found after I updated open-cluster-management.io/api to the latest commit in go.mod
Confirmed with @qiujian16, the `not found` issue will be fixed by https://github.com/open-cluster-management-io/addon-framework/pull/232.
I will update this PR then after the above PR is merged.

Comment (haoqing0110) [reply pkg/common/helpers/clusters_test.go:1]: @promid get it!

File: pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller_test.go [hunk 2]
```diff
@@ -1051,6 +1055,90 @@ func TestDeployOnKube111(t *testing.T) {
...
+		req := corev1.ResourceRequirements{}
+		actualRequests := req.String()
+		if actualResources != actualRequests {
+			t.Errorf("expect:\n%s\nbut got:\n%s", actualRequests, actualResources)
+		}
+	}
+}
+
+func TestRenderingResourceRequirements(t *testing.T) {
+	r := &operatorapiv1.ResourceRequirement{
```
Comment (qiujian16) [inline pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller_test.go:1110]: seems to me the three tests can be put in one test func with different cases. It also worth a test case with type is ResourceRequirement and ResourceRequirements is nil or empty

Comment (promid) [reply pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller_test.go]: merged into one unit test with 3 cases

**Comment by haoqing0110** [comment]
LGTM

**Comment by promid** [comment]
executed `make fmt-imports`

**Comment by qiujian16** [comment]
/approve
/lgtm

I am fine to merge this, I think we also need to add an integration test case as a followup
